app/javascript/react/components/capeform/NewCapeForm.js

```diff
@@ -37,7 +38,28 @@ const NewCapeForm = () => {
     return _.isEmpty(submitErrors);
   };
 
-  const addSuperhero = () => {
+  const onFormChange = event => {
+    const key = event.currentTarget.id;
+    const value = event.currentTarget.value;
+    setForm({
+      ...form,
+      [key]: value
+    });
+  };
+
+  const clearForm = () => {
+    setForm(defaultForm);
+  };
+
+  const formSubmit = event => {
+    event.preventDefault();
+    if (validForSubmission()) {
+      event.preventDefault();
+      addSuperhero();
+    }
+  };
+
+  function addSuperhero() {
```

Comment:
you should be consistent when defining methods. in all of your other methods you use an arrow function, but here you use the function keyword. Choose one and stick with it! (I recommend arrow)